master/reference/public-cloud/aws.md

```diff
@@ -76,3 +81,137 @@ you must configure outgoing NAT on your [Calico IP pool]({{site.baseurl}}/{{page
+Calico - if you are using an orchestrator such as Kubernetes, you'll need to follow
+the appropriate upgrade instructions for your orchestrator.  However, the instructions
+below may be partially applicable.
+
+#### Download calicoctl
+
+Ensure you have the latest version of `calicoctl` downloaded. See [releases page]({{site.baseurl}}/{{page.version}}/releases) 
+for the appropriate link.
+
+The `calicoctl` should be downloaded to each node and to your management server (if you use one).
+
+#### Upgrade your calico/node containers
+
+Upgrade your `calico/node` container instances on each host. See [releases page]({{site.baseurl}}/{{page.version}}/releases) 
+for details of the `calico/node` container version.
+
+Any calico/node instance that was running prior to v2.1 will have incorrect
+host subnet information configured (it will be /32 for the IPv4 management address).
+It is important that the subnet configuration is correct prior to using the `cross-subnet`
+IPIP feature.
+
+You can verify which of your nodes is correctly configured using calicoctl.
+
+Run `calicoctl get nodes --output=wide` to check the configuration.  e.g.
+
+```
+$ calicoctl get nodes --output=wide
+NAME    ASN       IPV4           IPV6   
+node1   (64512)   10.0.2.15/24          
+node2   (64512)   10.0.2.10/32          
+```
+
+In this example, node1 has the correct subnet information whereas node2 needs
+to be fixed.
+
+The subnet configuration may be fixed in a few different ways depending on how 
+you have deployed your calico/node containers.  We'll discuss the options below.
+
+**Configure the IP and subnet through runtime args/environments**
+
+The IP address may be explicitly specified using the `--ip` option on 
+`calicoctl node run` or the `IP` environment if you are starting the container
+directly.
+
+If you are currently specifying the IP address, you will need to update the
+parameter to include the network by specifying the IP and network in CIDR form (e.g. 
+`10.0.2.10/24`.
+
+For example (if using calicoctl node run):
+```
+calicoctl node run --ip 10.0.2.10/24
+```
+
+
+**Autodetect the IP and subnet**
+
+The node can be instructed to autodetect the IP address and subnet everytime it
+is restarted.  Use a value of `autodetect` for the IP address in the `--ip` option
+on `calicoctl node run` or the `IP` environment if you are starting the container
+directly.
+
+In addition, the `--ip-autodetection-method` argument or the `IP_AUTODETECT_METHOD`
+environment can be used to specify the method used to auto detect the host address 
+and subnet.  See [calico/node configuration guide]({{site.baseurl}}/{{page.version}}/reference/node/configuration)
+and [calicoctl command reference]({{site.baseurl}}/{{page.version}}/reference/calicoctl/commands/node/run)
+for details.
+
+For example (if using calicoctl node run):
+```
+calicoctl node run --ip autodetect --ip-autodetection-method can-reach=8.8.8.8
```

Comment:
wouldn't it be better to add an `=` between `method` and `can-reach` like --ip-autodetection-method=can-reach=8.8.8.8` so it doesn't look like `can-reach=8.8.8.8` is a separate flag. + it would be consistent with our ip autodetection docs